dev/com.ibm.ws.webcontainer/src/com/ibm/ws/webcontainer/osgi/mbeans/PluginGenerator.java

```diff
@@ -885,85 +867,48 @@ protected synchronized void generateXML(String rootLoc, String serverName,
     }
 
     @FFDCIgnore(IOException.class)
-    public static void copyFile(File in, File out)
-                    throws IOException
-                {
-                    FileChannel inChannel = new
-                        FileInputStream(in).getChannel();
-                    FileChannel outChannel = new
-                        FileOutputStream(out).getChannel();
-                    try {
-                        inChannel.transferTo(0, inChannel.size(),
-                                outChannel);
-                    }
-                    catch (IOException e) {
-                        throw e;
-                    }
-                    finally {
-                        if (inChannel != null) inChannel.close();
-                        if (outChannel != null) outChannel.close();
-                    }
-                }
+    public static void copyFile(File in, File out) throws IOException {
+        FileChannel inChannel = new FileInputStream(in).getChannel();
+        FileChannel outChannel = new FileOutputStream(out).getChannel();
+        try {
+            inChannel.transferTo(0, inChannel.size(),
+                                 outChannel);
+        } catch (IOException e) {
+            throw e;
+        } finally {
+            if (inChannel != null)
+                inChannel.close();
+            if (outChannel != null)
+                outChannel.close();
+        }
+    }
 
-    private static TransformerFactory getTransformerFactory() {
+    public static TransformerFactory getTransformerFactory() {
         if (TraceComponent.isAnyTracingEnabled() && tc.isEntryEnabled()) {
-            Tr.entry(tc, "getTransformerFactory", "CHANGE_TRANSORMER = " + CHANGE_TRANSFORMER);
+            Tr.entry(tc, "getTransformerFactory");
         }
 
         TransformerFactory tf = null;
 
-        if (CHANGE_TRANSFORMER) {
-
-            // Synchronize setting and restoring the jvm property to prevent this sequence:
-            // 1. Thread 1 gets jvm property
-            // 2. Thread 1 sets jvm property
-            // 3. Thread 2 gets jvm property set by Thread 1
-            // 4. Thread 1 resets jvm property to value obtained at 1.
-            // 5. Thread 2 resets jvm property to value set by Thread 1.
-            synchronized (transformerLock) {
+        //We want to use the the Interpretive processor (Apache Xalan) but need to work out which one
+        //This is either going to be the one package with an IBM Java SDK or the one packaged as a fallback by Oracle
+        boolean useApacheXalanTransformer = JavaInfo.isAvailable(IBM_XLTXEJ_COMPILED_TRANSFORMER_FACTORY_CLASS_NAME) &&
+                                    JavaInfo.isAvailable(XALAN_TRANSFORMER_FACTORY_CLASS_NAME) &&
+                                    JavaInfo.isAvailable(SAX_LEXICAL_HANDLER_CLASS_NAME);
```

Comment:
Comment needs updating I think?

IIRC, if we see the XLTXEJ transformer and the XALAN transformer are both available, we need to manually select the XALAN transformer, as we will hit problems with XLTXEJ?

Also unsure if the check for `SAX_LEXICAL_HANDLER_CLASS_NAME` is actually necessary.